components/accelerator/nvidia/hw-slowdown/component.go

```diff
@@ -32,201 +31,253 @@ const (
+	d.suggestedActions = &common.SuggestedActions{
+		RepairActions: []common.RepairActionType{
+			common.RepairActionTypeHardwareInspection,
+		},
+		Descriptions: []string{
+			"Hardware slowdown are often caused by GPU overheating or power supply unit (PSU) failing, please do a hardware inspection to mitigate the issue",
 		},
-	}, nil
+	}
 }
 
-// summarizeDataSources summarizes data source information
-func summarizeDataSources(events []components.Event) string {
-	dataSources := make(map[string]int)
-	for _, event := range events {
-		if event.ExtraInfo == nil {
-			continue
-		}
+type Data struct {
+	ClockEvents []nvidianvml.ClockEvents `json:"clock_events,omitempty"`
 
-		src := event.ExtraInfo["data_source"]
-		if src != "" {
-			dataSources[src]++
-		}
-	}
+	// timestamp of the last check
+	ts time.Time `json:"-"`
+	// error from the last check
+	err error `json:"-"`
 
-	dsDescs := make([]string, 0, len(dataSources))
-	for ds, count := range dataSources {
-		dsDescs = append(dsDescs, fmt.Sprintf("%s: %d", ds, count))
-	}
-	return strings.Join(dsDescs, ", ")
+	// tracks the healthy evaluation result of the last check
+	healthy bool `json:"-"`
+	// tracks the reason of the last check
+	reason string `json:"-"`
+	// tracks the suggested actions of the last check
+	suggestedActions *common.SuggestedActions `json:"-"`
 }
 
-func (c *component) Events(ctx context.Context, since time.Time) ([]components.Event, error) {
-	return c.eventBucket.Get(ctx, since)
-}
-
-func (c *component) Metrics(ctx context.Context, since time.Time) ([]components.Metric, error) {
-	log.Logger.Debugw("querying metrics", "since", since)
-
-	hwSlowdown, err := nvidia_query_metrics_clock.ReadHWSlowdown(ctx, since)
-	if err != nil {
-		return nil, err
-	}
-	hwSlowdownThermal, err := nvidia_query_metrics_clock.ReadHWSlowdownThermal(ctx, since)
-	if err != nil {
-		return nil, err
-	}
-	hwSlowdownPowerBrake, err := nvidia_query_metrics_clock.ReadHWSlowdownPowerBrake(ctx, since)
-	if err != nil {
-		return nil, err
+func (d *Data) getError() string {
+	if d == nil || d.err == nil {
+		return ""
 	}
+	return d.err.Error()
+}
 
-	ms := make([]components.Metric, 0, len(hwSlowdown)+len(hwSlowdownThermal)+len(hwSlowdownPowerBrake))
-	for _, m := range hwSlowdown {
-		ms = append(ms, components.Metric{
-			Metric: m,
-			ExtraInfo: map[string]string{
-				"gpu_id": m.MetricSecondaryName,
-			},
-		})
-	}
-	for _, m := range hwSlowdownThermal {
-		ms = append(ms, components.Metric{
-			Metric: m,
-			ExtraInfo: map[string]string{
-				"gpu_id": m.MetricSecondaryName,
-			},
-		})
-	}
-	for _, m := range hwSlowdownPowerBrake {
-		ms = append(ms, components.Metric{
-			Metric: m,
-			ExtraInfo: map[string]string{
-				"gpu_id": m.MetricSecondaryName,
+func (d *Data) getStates() ([]components.State, error) {
+	if d == nil {
+		return []components.State{
+			{
+				Name:    Name,
```

Comment:
@cardyok This used to be `hw_slowdown`, now `accelerator-nvidia-hw-slowdown`

Reply:
got it, currently we do not have dependency